src/image/loading_displaying.js

```diff
@@ -280,6 +280,79 @@ function _createGif(
   finishCallback();
 }
 
+/**
+ * @private
+ * @param {Constant} xAlign either LEFT, RIGHT or CENTER
+ * @param {Constant} yAlign either TOP, BOTTOM or CENTER
+ * @param {Number} dx
+ * @param {Number} dy
+ * @param {Number} dw
+ * @param {Number} dh
+ * @param {Number} sw
+ * @param {Number} sh
+ * @returns {Object}
+ */
+
+function _imageContain(xAlign, yAlign, dx, dy, dw, dh, sw, sh) {
+  const r = Math.max(sw / dw, sh / dh, 1);
```

Comment:
Is the `Math.max(..., 1)` there to prevent the image from stretching past its original pixel size? If so, I think we can take it out, because the intended behaviour is to allow images to uniformly scale possibly above its original dimensions to fit or fill the container.

Reply:
It calculates, the minimum needed ratio that we need to scale the destination box by, so that we don't distort the image aspect ratio. Let me, know if it still confuses you.